### PR TITLE
rgw: fix bugs in listobjectsv1

### PR DESCRIPTION
**Pull request by albIN7**
Fixes: https://tracker.ceph.com/issues/40663

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by albIN7** [comment]
@cbodley please review

**Comment by mkogan1** [comment]
@albIN7 Applying the commit there is an assertion abort on radosgw start:
```
$ ./bin/radosgw -f -c ./ceph.conf --log-file=./out/radosgw.8000.log --admin-socket=./out/radosgw.8000.asok --pid-file=./out/radosgw.8000.pid -n client.rgw '--rgw_frontends=beast port=8000'
2019-07-08T12:03:30.431-0400 7f673ce358c0 -1 WARNING: all dangerous and experimental features are enabled.
2019-07-08T12:03:30.446-0400 7f673ce358c0 -1 WARNING: all dangerous and experimental features are enabled.
2019-07-08T12:03:30.447-0400 7f673ce358c0 -1 WARNING: all dangerous and experimental features are enabled.
/mnt/osd--host/src-git/ceph--master--dectris_sync/src/common/Formatter.cc: In function 'virtual void ceph::XMLFormatter::close_section()' thread 7f65e6ffd700 time 2019-07-08T12:03:30.599077-0
400
/mnt/osd--host/src-git/ceph--master--dectris_sync/src/common/Formatter.cc: 416: FAILED ceph_assert(!m_sections.empty())
 ceph version v15.0.0-2526-gf27bd45307 (f27bd45307d2c20ae10bd0c94b6801b15d1b5bab) octopus (dev) 
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x19d) [0x7f673f32cc1f]
 2: (()+0x1541ea1) [0x7f673f32cea1]
 3: (ceph::XMLFormatter::close_section()+0x45) [0x7f673f266d73]
 4: (RGWListBucket_ObjStore_S3::send_versioned_response()+0xb77) [0x55b6c1baa567]
 5: (RGWListBucket_ObjStore_S3::send_response()+0xe7) [0x55b6c1baaa33]
 6: (RGWOp::complete()+0x23) [0x55b6c15387dd]
 7: (rgw_process_authenticated(RGWHandler_REST*, RGWOp*&, RGWRequest*, req_state*, bool)+0x15aa) [0x55b6c16e7f3a]
 8: (process_request(RGWRados*, RGWREST*, RGWRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rgw::auth::StrategyRegistry const&, RGWRestfulIO
*, OpsLogSocket*, optional_yield, rgw::dmclock::Scheduler*, int*)+0x1484) [0x55b6c16e9bfc]
 9: (()+0xd57808) [0x55b6c1599808]
 10: (()+0xd533ae) [0x55b6c15953ae]
 11: (()+0xd5ec6e) [0x55b6c15a0c6e]
 12: (()+0xd5e92c) [0x55b6c15a092c]
 13: (()+0xd5e653) [0x55b6c15a0653]
 14: (make_fcontext()+0x2f) [0x55b6c1d678ef]

```

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -951,6 +940,7 @@ void RGWListBucket_ObjStore_S3::send_versioned_response()
   s->formatter->close_section();
   rgw_flush_formatter_and_reset(s, s->formatter);
 }
+  s->formatter->close_section();
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: i think this is crashing because it comes after the call to `rgw_flush_formatter_and_reset()`

Comment (albIN7) [reply src/rgw/rgw_rest_s3.cc]: Makes sense. But the multisite tests are still failing

Comment (albIN7) [reply src/rgw/rgw_rest_s3.cc]: @cbodley Works fine now. Could you please check

**Comment by mkogan1** [comment]
LGTM, Tested OK after the latest update.

**Comment by albIN7** [comment]
> LGTM, Tested OK in multi-site after the latest update.

Thanks @mkogan1

**Comment by cbodley** [comment]
thanks @albIN7, nice work

**Pull request merged**